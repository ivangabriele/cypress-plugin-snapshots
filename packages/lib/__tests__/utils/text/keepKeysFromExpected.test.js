describe("utils/keepKeysFromExpected", () => {
	const expected = {
		field: "value",
		undefined: undefined,
		null: null,
		items: [
			{ name: "item1", items: [{ name: "subitem1" }] },
			{ name: "item2" },
			{ name: "item3" },
			{ name: null },
			{ name: undefined },
			null,
			undefined,
		],
	}

	const actual = {
		field: "value",
		undefined: undefined,
		null: null,
		extra: "extra",
		extraUndefined: undefined,
		extraNull: null,
		items: [
			{
				name: "item1",
				items: [
					{ name: "subitem1", extra: "extra1" },
					{
						name: "subitem2",
						extra: "extra2",
						undefined: undefined,
						null: null,
						extraUndefined: undefined,
						extraNull: null,
					},
					null,
					undefined,
				],
			},
			{ name: "item2" },
			{ name: "item3" },
			{ name: null },
			{ name: undefined },
			null,
			undefined,
			{ name: "item4" },
			{ name: null },
			{ name: undefined },
		],
	}

	it("ignoreExtraFields: false, ignoreExtraArrayItems: false", () => {
		const config = {
			ignoreExtraFields: false,
			ignoreExtraArrayItems: false,
		}
		const keepKeysFromExpected = require("../../../src/utils/text/keepKeysFromExpected")
		const result = keepKeysFromExpected(actual, expected, config)
		expect(result).toMatchSnapshot()
	})

	it("ignoreExtraFields: true, ignoreExtraArrayItems: false", () => {
		const config = {
			ignoreExtraFields: true,
			ignoreExtraArrayItems: false,
		}
		const keepKeysFromExpected = require("../../../src/utils/text/keepKeysFromExpected")
		const result = keepKeysFromExpected(actual, expected, config)
		expect(result).toMatchSnapshot()
	})

	it("ignoreExtraFields: true, ignoreExtraArrayItems: true", () => {
		const config = {
			ignoreExtraFields: true,
			ignoreExtraArrayItems: true,
		}
		const keepKeysFromExpected = require("../../../src/utils/text/keepKeysFromExpected")
		const result = keepKeysFromExpected(actual, expected, config)
		expect(result).toMatchSnapshot()
	})

	it("ignoreExtraFields: false, ignoreExtraArrayItems: true", () => {
		const config = {
			ignoreExtraFields: false,
			ignoreExtraArrayItems: true,
		}
		const keepKeysFromExpected = require("../../../src/utils/text/keepKeysFromExpected")
		const result = keepKeysFromExpected(actual, expected, config)
		expect(result).toMatchSnapshot()
	})
})
