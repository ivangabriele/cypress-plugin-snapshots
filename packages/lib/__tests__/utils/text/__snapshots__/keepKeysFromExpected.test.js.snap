// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: false 1`] = `
Object {
  "extra": "extra",
  "extraNull": null,
  "extraUndefined": undefined,
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "extraNull": null,
          "extraUndefined": undefined,
          "name": "subitem2",
          "null": null,
          "undefined": undefined,
        },
        null,
        undefined,
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
    null,
    undefined,
    Object {
      "name": "item4",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
  ],
  "null": null,
  "undefined": undefined,
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: true 1`] = `
Object {
  "extra": "extra",
  "extraNull": null,
  "extraUndefined": undefined,
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
    null,
    undefined,
  ],
  "null": null,
  "undefined": undefined,
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: false 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "extraNull": null,
          "extraUndefined": undefined,
          "name": "subitem2",
          "null": null,
          "undefined": undefined,
        },
        null,
        undefined,
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
    null,
    undefined,
    Object {
      "name": "item4",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
  ],
  "null": null,
  "undefined": undefined,
}
`;

exports[`utils/keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: true 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": null,
    },
    Object {
      "name": undefined,
    },
    null,
    undefined,
  ],
  "null": null,
  "undefined": undefined,
}
`;
